src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1841,14 +1841,15 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                                         expressionStatement.Expression is BoundCall { Method: var initMethod } && initMethod.IsDefaultValueTypeConstructor();
                                 }
 
-                                return body;
                             }
                             else
                             {
                                 Debug.Assert(constructor.Initializer is null);
                                 Debug.Assert(constructor.Locals.IsEmpty);
+                                Debug.Assert(BindImplicitConstructorInitializerIfAny(method, compilationState, diagnostics) is null);
```

Comment:
>Debug.Assert(BindImplicitConstructorInitializerIfAny(method, compilationState, diagnostics) is null); [](http://example.com/codeflow?start=32&length=101)

Do we really need this assert? It looks like it is a source of unwanted (duplicate) diagnostics.

Reply:
The early return is only safe so long as there's no implicit constructor returned, so I'd prefer to keep it. I'll use a discarded diagnostics bag.